src/impl/icetransport.cpp

```diff
@@ -154,6 +133,32 @@ IceTransport::IceTransport(const Configuration &config, candidate_callback candi
 	mAgent = decltype(mAgent)(juice_create(&jconfig), juice_destroy);
 	if (!mAgent)
 		throw std::runtime_error("Failed to create the ICE agent");
+
+	// Add TURN servers
+	int k = 0;
+	for (auto &server : servers) {
+		addTURNServer(server);
```

Comment:
You should filter out STUN servers explicitly.

Reply:
Done! New method returns early on STUN servers